catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-converter/src/main/java/org/codice/ddf/spatial/ogc/wfs/v1_1_0/catalog/converter/impl/FeatureCollectionConverterWfs11.java

```
@@ -0,0 +1,209 @@
+import javax.xml.namespace.QName;
+import org.codice.ddf.spatial.ogc.catalog.common.converter.XmlNode;
+import org.codice.ddf.spatial.ogc.wfs.catalog.common.WfsConstants;
+import org.codice.ddf.spatial.ogc.wfs.catalog.common.WfsFeatureCollection;
+import org.codice.ddf.spatial.ogc.wfs.catalog.common.WfsQnameBuilder;
+import org.codice.ddf.spatial.ogc.wfs.catalog.converter.FeatureConverter;
+import org.codice.ddf.spatial.ogc.wfs.v1_1_0.catalog.common.Wfs11Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class works in conjunction with XStream to convert a {@link WfsFeatureCollection} to XML
+ * according to the GML 3.1.1 spec. It will also convert respective XML into a {@link
+ * WfsFeatureCollection}.
+ */
+public class FeatureCollectionConverterWfs11 implements Converter {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(FeatureCollectionConverterWfs11.class);
+
+  private static final String FEATURE_MEMBER_ELEMENT = "featureMember";
+
+  private static final String FEATURE_MEMBERS_ELEMENT = "featureMembers";
+
+  private Map<String, String> prefixToUriMapping = new HashMap<>();
+
+  private String contextRoot;
+
+  private Map<String, FeatureConverter> featureConverterMap = new HashMap<>();
+
+  public FeatureCollectionConverterWfs11() {
+    prefixToUriMapping.put(Wfs11Constants.XSI_PREFIX, XMLConstants.W3C_XML_SCHEMA_INSTANCE_NS_URI);
+    prefixToUriMapping.put(Wfs11Constants.WFS_NAMESPACE_PREFIX, Wfs11Constants.WFS_NAMESPACE);
+    prefixToUriMapping.put(Wfs11Constants.GML_PREFIX, Wfs11Constants.GML_3_1_1_NAMESPACE);
+  }
+
+  @Override
+  public boolean canConvert(Class clazz) {
+    if (!WfsFeatureCollection.class.isAssignableFrom(clazz)) {
+      LOGGER.debug("Cannot convert: {}", clazz.getName());
+    }
+    return WfsFeatureCollection.class.isAssignableFrom(clazz);
+  }
+
+  @Override
+  public void marshal(Object value, HierarchicalStreamWriter writer, MarshallingContext context) {
```

❗️ @glenhein isn't this taking a list of metacards and creating a WFS feature collection?  I can't see a reason why would we need this.  I think we only need the unmarshal.

---

it was a carry-over from the other impls. But I can look at removing it.

---

@jlcsmith The marshal method is required by the Converter interface. I could just have it throw a runtime exception with a "Not Implemented" message. Is that OK, or should I leave it as-is?

---

I'd vote to have it throw an UnsupportedOperationException.